### PR TITLE
feat(export): remove EpisodeExport — single Episode path

### PR DESCRIPTION
**Pull request by felixmaximilian**
Stacked on #230.

Port of PR #219 onto `refactor/timestep-ordering-v2`. The core idea is identical but several things needed adaptation given that #230 had already restructured `Episode` significantly.

## What this does

- **Delete `EpisodeExport`** — `EpisodeBuilder.forward` now always returns `Episode` (TensorClass). The `torch.export.register_dataclass` / `pytree.register_dataclass` registration and `timestep_keys` drop-field machinery are all gone.
- **`TensorDict.from_dict` → direct constructor** — `from_dict` goes through `from_any → _is_dataclass → hasattr` which Dynamo cannot trace. Replaced with `TensorDict(d, batch_size=[b, t])` everywhere in `forward()`. Same fix for `Index`: `Index.from_tensordict(TensorDict(index, batch_size=[t]))`.
- **Unify `PolicyObjective`** — remove `@overload`, `isinstance(episode, EpisodeExport)` branch, and `tree_map_with_path` path. Single `Episode._compute_logits` path.
- **`ControlTransformer.forward`** — remove `is_exporting()` guard; always return `TensorDict`.
- **`norm.py`** — remove `is_exporting()` / `ValueError` split from `Scaler` and `UniformBinner`; clamp at output level instead of clamping the raw input first.
- **`test_export.py`** — `episode_export` fixture returns `Episode`; `test_episode` uses a shared `ep_dict` helper; `test_torch_export_fake` zips leaves by position (no `asdict`/`is_dataclass`); policy-continuous tolerance hack removed (single code path → bit-exact); `test_onnx_inference` added (ORT vs eager numerical regression).
- **`CLAUDE.md`** — new file documenting export architecture and remaining guards.

## What had to be done differently vs PR #219

**`Episode` already had `embeddings` (not `projected_embeddings` + `role_embeddings`)** — #230 pre-computed the sum and dropped both fields. PR #219 still had them separate and used `.apply(torch.add, ...)` for ONNX safety. Not needed here since `embeddings` is already computed via `tree_map` (not `TensorDict.__add__`).

**`EpisodeExport.token_embeddings`** — in PR #219, `EpisodeExport` still had `projected_embeddings` / `role_embeddings`. In #230, `EpisodeExport` had `token_embeddings` as a property backed by `timestep_keys`. Removing `EpisodeExport` entirely resolves all of this: `Episode.token_embeddings` is the single property.

**`timestep_keys` drop-field gymnastics not needed** — the entire `pytree.register_dataclass(drop_field_names=["timestep_keys"])` + `_timestep_keys` precomputed in `__init__` to avoid `AttrSource` chains is eliminated since we no longer export `EpisodeExport`.

**`logger.warning` kept with `is_compiling()` guard** — PR #219 removed it entirely. Since the attention mask cache warm-up is still required before `torch.export`, the warning remains useful. The `not torch.compiler.is_compiling()` guard from #230 keeps it safe.

**`norm.py` clamping** — PR #219 added `input = torch.clamp(input, ...)` before computation. Applied at output level instead (`Scaler`: `.clamp(out_min, out_max)`; `UniformBinner`: `.clamp(0, bins-1)`), which is semantically equivalent and avoids touching the input twice.

## Test plan
- [x] 62 tests pass including all `test_torch_export`, `test_onnx_export`, and new `test_onnx_inference`
- [ ]

### CODE REVIEW

File: src/rmind/components/episode.py [hunk 1]
```diff
@@ -157,15 +124,14 @@ def attention_mask_cache_is_warm(self) -> bool:
...
 
-        (b, t), device = mit.one({
-            (leaf.shape[:2], leaf.device)
-            for leaf in tree_leaves(input_tokens)
-            if leaf is not None
-        })
+        first_leaf = mit.first(
+            leaf for leaf in tree_leaves(input_tokens) if leaf is not None
+        )
+        b, t, device = first_leaf.shape[0], first_leaf.shape[1], first_leaf.device
```
Comment (egorchakov) [inline src/rmind/components/episode.py]: do we still need to use pytree API here? we used to just take `batch_size` and `device` from `input` (https://github.com/yaak-ai/rmind/pull/161/changes#diff-de9dfb0b07d8d939aff21f495a6e0da1a949db7d9ecbcc7eb9a69743011320ccL199), perhaps that's export-friendly now?

ditto for `_build_index` and `_build_role_embeddings`

i guess the general idea is to keep things TensorDict as much as possible, and only drop down to pytree in case of export issues (thoughts @koritsky @sandhawalia ?)

File: src/rmind/utils/tensordict_export_patch.py
Comment (egorchakov) [inline src/rmind/utils/tensordict_export_patch.py:1]: https://github.com/pytorch/tensordict/pull/1704 got merged -- should we switch to https://pypi.org/project/tensordict-nightly/ until the next https://pypi.org/project/tensordict/ release?

File: tests/test_export.py [hunk 2]
```diff
@@ -8,19 +7,35 @@
...
 if TYPE_CHECKING:
     from tests.conftest import EmbeddingDims
 
 
+@pytest.fixture(autouse=True, scope="session")
+def _soft_pending_unbacked() -> None:
+    # tensordict's _device_recorder and _CONSTRUCTORS are global dicts mutated
+    # during export tracing (dynamo side-effect). This creates a spurious
+    # "pending unbacked symbol u0" error even though the exported graph is valid.
+    # Demote to warning so torch.export tests are not falsely blocked.
```
Comment (egorchakov) [inline tests/test_export.py:32]: worth opening a tensordict issue?

File: tests/test_export.py [hunk 3]
```diff
@@ -118,16 +131,17 @@ def control_transformer(
...
+    def ep_dict(src: Episode) -> dict:
+        d = src.to_dict() | {
+            "embeddings": src.embeddings.to_dict(),
+            "token_embeddings": src.token_embeddings,
+        }
+        d.pop("timestep_keys", None)
+        return d
+
+    episode_dict = ep_dict(episode)
+    episode_export_dict = ep_dict(episode_export)
```
Comment (egorchakov) [inline tests/test_export.py]: since `episode` and `episode_export` are both `Episode` (i.e. `TensorClass`), could we just do direct comparison without tree stuff?

**Review by egorchakov — CHANGES_REQUESTED** [review]

File: src/rmind/components/episode.py (hunk 1 above)
Comment (koritsky) [reply src/rmind/components/episode.py]: yes, good reference

also, is batch still a `TensorTree` or we now operate with `TensorDict`s?

Comment (egorchakov) [reply src/rmind/components/episode.py]: this export thing runs deep 

as part of https://github.com/yaak-ai/rmind/pull/161 we [changed the dataloader `collate_fn`](https://github.com/yaak-ai/rmind/pull/161/changes#diff-a60a7844e4f06be9728f2f9d35226ece78e34c561cafd81f2a33d8121efdbce0) so each `rbyte.Batch` becomes a `dict`/`TensorTree`, hence the expected input type

can prob be reverted back to `batch: TensorDict`, though some things like `rmind.components.containers.ModuleDict` (in this case `tokenizers`, `embeddings`, `projections`) still expect pytrees as input and need to be converted (back) to TensorDict

File: tests/test_export.py
Comment (egorchakov) [inline tests/test_export.py:1]: afaict as of this PR we only rely on `torch.compiler.is_exporting()` to cache attention masks

do we still need `test_torch_export_fake` (and any upstream fixtures)?

**Review by egorchakov — CHANGES_REQUESTED** [review]

File: src/rmind/components/episode.py (hunk 1 above)
Comment (koritsky) [reply src/rmind/components/episode.py]: i thought about going back to `TensorDict` exactly because `dict/TensorTree` structure was introduced for exportability in #161. since now we simplify things, probably it's a good time to revert back, but ofc only if it's compatible with `torch.export`

@felixmaximilian prob worth giving this to claude and see if it turns out well

File: tests/test_export.py
Comment (felixmaximilian) [reply tests/test_export.py:1]: Indeed, let's maybe merge this PR first. Claude provided a draft on full Tensordict migration. It's not that big and complicated in the end. But there batch is a tensordict through out the call graph.
https://github.com/yaak-ai/rmind/pull/237

File: src/rmind/components/episode.py [hunk 4]
```diff
@@ -135,75 +113,54 @@ def attention_mask_cache_is_warm(self) -> bool:
...
+        input_tokens.update({
+            modality: {
+                name: torch.tensor(values, device=device).expand(b, t, -1)
+                for name, values in names.items()
+            }
+            for modality, names in self.special_tokens.items()
+        })
         input_embeddings = self.embeddings(input_tokens)
         projected_embeddings = self.projections(input_embeddings)
+        projected_td = TensorDict(
```
Comment (egorchakov) [inline src/rmind/components/episode.py:134]: [nit/naming] i think just appending `_td` is slightly more readable, like `projected_embeddings_td`

also for `role_td`

File: tests/test_export.py (hunk 3 above)
Comment (egorchakov) [reply tests/test_export.py]: assuming gone in https://github.com/yaak-ai/rmind/pull/237

**Review by egorchakov — APPROVED** [review]
